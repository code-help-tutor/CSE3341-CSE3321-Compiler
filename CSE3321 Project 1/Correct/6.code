
program int X , Y, Z;

        input X, Y, Z;
		
		while not (X := 0) begin
		
			output Y;
		
			while Y < 10 begin
				
				Y := Y + 1;
				
			;
			
			output Y;
			
			Y := Z
			
			X := X - 1;
			
		;
		
		
end  
      
