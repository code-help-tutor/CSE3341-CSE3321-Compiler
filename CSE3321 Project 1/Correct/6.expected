ID[program]
ID[int]
ID[X]
COMMA
ID[Y]
COMMA
ID[Z]
SEMICOLON
ID[input]
ID[X]
COMMA
ID[Y]
COMMA
ID[Z]
SEMICOLON
WHILE
NOT
LPAREN
ID[X]
ASSIGN
CONST[0]
RPAREN
BEGIN
ID[output]
ID[Y]
SEMICOLON
WHILE
ID[Y]
LESS
CONST[10]
BEGIN
ID[Y]
ASSIGN
ID[Y]
ADD
CONST[1]
SEMICOLON
SEMICOLON
ID[output]
ID[Y]
SEMICOLON
ID[Y]
ASSIGN
ID[Z]
ID[X]
ASSIGN
ID[X]
SUBTRACT
CONST[1]
SEMICOLON
SEMICOLON
END
